src/test/java/uk/gov/nationalarchives/pdi/step/jena/serializer/JenaSerializerStepIT.java

```diff
@@ -0,0 +1,192 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+public class JenaSerializerStepIT {
+    @BeforeAll
+    public static void setup() throws KettleException {
+        KettleClientEnvironment.init();
+    }
+
+    @Test
+    public void resolves_jena_model_field_variable() throws KettleException {
+        final JenaSerializerStepMeta meta = getMeta();
+        final StepDataInterface data = getData();
+        final StepMockHelper helper = mockHelper();
+        final JenaSerializerStep step = mockStep(helper);
+
+        step.setVariable("jenaModelFieldVar", "model");
+        meta.setJenaModelField("${jenaModelFieldVar}");
+
+        final boolean rowProcessed = step.processRow(meta, data);
+
+        // Would throw if var was not resolved
+        assertTrue(rowProcessed);
+    }
+
+    @Test
+    public void resolves_filename_variable() throws KettleException {
+        // TODO(SL): Use temp folder instead of random file
+        final String expectedFilename = String.valueOf(new Random().nextInt());
+
+        final JenaSerializerStepMeta meta = getMeta();
+        final StepDataInterface data = getData();
+        final StepMockHelper helper = mockHelper();
+        final JenaSerializerStep step = mockStep(helper);
+
+        meta.setJenaModelField("model");
+
+        step.setVariable("filenameVar", expectedFilename);
+        meta.setFileDetail(
+                new JenaSerializerStepMeta.FileDetail() {{
+                    filename = "${filenameVar}";
+                }});
+
+        // First pass create model, second serialise
+        step.processRow(meta, data);
+        step.processRow(meta, data);
+
+        File f = new File(expectedFilename);
+
+        assertTrue(f.exists() && !f.isDirectory());
+    }
+
+    @Test
+    public void resolves_serialization_format_variable() throws KettleException {
+        final JenaSerializerStepMeta meta = getMeta();
+        final StepDataInterface data = getData();
+        final StepMockHelper helper = mockHelper();
+        final JenaSerializerStep step = mockStep(helper);
+
+        meta.setJenaModelField("model");
+        meta.setFileDetail(new JenaSerializerStepMeta.FileDetail());
+
+        step.setVariable("serialisationFormatVar", "RDF/XML");
+        meta.setSerializationFormat("${serialisationFormatVar}");
+
+        // First pass create model, second serialise
+        step.processRow(meta, data);
+        final boolean rowProcessed = step.processRow(meta, data);
+
+        // Would throw if var was not resolved
+        assertFalse(rowProcessed);
+    }
+
+    private StepDataInterface getData() {
+        return new JenaSerializerStepData() {{

```

Comment:
Why `{{`? This is generally not a good idea... Better to create and then call `init()` please

Reply:
Done.
Why?

Reply:
So the following using the `{{` syntax:

```
new JenaSerializerStepData() {{ init(); }}
```

This actually creates a new anonymous sub-class of `JenaSerializerStepData` that inherits from `JenaSerializerStepData`. The constructor of the anonymous sub-class, will call `init()` on the anonymous sub-class, which effectively calls it on the super-class `JenaSerializerStepData`. Finally you create an instance of the anonymous sub-class with `new`.

Whereas the following code:
```
JenaSerializerStepData jssd = new JenaSerializerStepData();
jssd.init();
```

Constructs an instance of the class `JenaSerializerStepData`, and then calls its `init()` function.

----

1. There is no good reason that you need to create an anonymous sub-classes here. I understand you did it for code-brevity, but there is no technical reason for it, e.g. you don't need to add methods or override methods.
2. Also at debug time, the identity of an anonymous sub-class can be quite confusing.

p.s. You are only the second developer I have seen write `{{` most Java developers are not even aware of its existence or what it does. Many that I have explained it to think that it is a syntactic mistake ;-)

Reply:
Perfect, thx.